plugins/powershell/unit_test/test_powershell_string.py

```diff
@@ -0,0 +1,54 @@
+import sys
+import os
+
+sys.path.append(os.path.abspath("../"))
+
+from unittest import TestCase
+from unittest.mock import patch
+from unit_test.util import Util
+from komand_powershell.actions.powershell_string import PowershellString
+
+
+class TestPowershellString(TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.params = {
+            "address": "10.0.1.11",
+            "username_and_password": {"password": "example_password", "username": "example_user"},
+            "host_name": "windows",
+            "script": "Get-Date",
+            "secret_key": {"secretKey": "s1e2c3r4e5t67k8e9y"},
+        }
+
+    @patch("subprocess.Popen", side_effect=Util.mock_powershell)
+    def test_powershell_string_local(self, mock_powershell):
+        params = Util.read_file_to_dict("inputs/local_connection.json.resp")
+        action = Util.default_connector(PowershellString(), params)
+        actual = action.run(params=self.params)
+        expected = {"stdout": "\nWednesday, January 12, 2022 1:36:51 PM\n\n\n", "stderr": ""}
+        self.assertEqual(actual, expected)
+
+    @patch("winrm.Session", side_effect=Util.mock_powershell)
+    def test_powershell_string_ntlm(self, mock_powershell):
+        params = Util.read_file_to_dict("inputs/ntlm_connection.json.resp")
+        action = Util.default_connector(PowershellString(), params)
+        actual = action.run(params=self.params)
+        expected = {"stdout": "\r\nThursday, January 13, 2022 2:44:55 AM\r\n\r\n\r\n", "stderr": ""}
+        self.assertEqual(actual, expected)
+
+    @patch("winrm.Session", side_effect=Util.mock_powershell)
+    def test_powershell_string_credssp(self, mock_powershell):
+        params = Util.read_file_to_dict("inputs/credssp_connection.json.resp")
+        action = Util.default_connector(PowershellString(), params)
+        actual = action.run(params=self.params)
+        expected = {"stdout": "\r\nThursday, January 13, 2022 3:18:12 AM\r\n\r\n\r\n", "stderr": ""}
+        self.assertEqual(actual, expected)
```

Comment:
I'm not sure if you mean this kind of parameterization but I've implemented it. It was possible only for 2 tests, because sue the rest of them are mocking different methods.